### PR TITLE
trello-41: journal landing pages

### PR DESCRIPTION
**Pull request by rnathanday**
This pull request implements public journal landing pages, and incorporates all feedback given on the landing page prototype.

There are both junit and selenium tests included. The junit tests (dspace/modules/journal-landing/journal-landing-webapp/src/test/java/org/datadryad/api/DryadJournalTest.java) run as part of the Travis build, and the selenium tests may be run from the `dryad-repo` directory using this command:
```
~/dryad-repo/selenium $ mvn package -DseleniumTestURL="http://192.168.111.223:9999" -Dtest=JournalLandingPageTests
``` 
PR #887 (rnathanday:journal-landing-new-metadata-fields) adds two journal metadata fields (journal.description, journal.memberName) that are used on a landing page, so that PR is a dependency of this one, although not a hard dependency as missing metadatavalues are not an error.

Documentation available in the wiki:
http://wiki.datadryad.org/Journal_Landing_Pages
http://wiki.datadryad.org/Journal_Landing_Pages_build
http://wiki.datadryad.org/Journal_Landing_Page_Mockups

### CODE REVIEW

File: dspace/config/dspace.cfg [hunk 1]
```diff
@@ -2472,6 +2472,7 @@ solr.dryad.server = ${default.solr.dryad.server}
 #-----------------SOLR SERVER CONFIGURATION---------------------#
 #---------------------------------------------------------------#
 solr.search.server = ${default.solr.search.server}
+solr.stats.server = ${default.solr.stats.server}
```
Comment (ryscher) [inline dspace/config/dspace.cfg:2475]: @daisieh is this the same parameter we just removed?

Comment (daisieh) [inline dspace/config/dspace.cfg:2475]: Nope, that was `solr.authority.server`

**Comment by ryscher** [comment]
#### Code Review Checklist
- [x] Read through the code to ensure it is:
- [x] Legible
- [x] Well-commented
- [x] Free of any "code smells"
- [x] Ensure the associated documentation is up to date.
- [x] Compile it on your local machine and test the new functionality.
- [x] Verify that the change is accompanied by adequate testing code. Verify that the code passes the tests.

File: dspace/modules/journal-landing/journal-landing-webapp/src/main/java/org/dspace/app/xmlui/aspect/journal/landing/ValidateRequest.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    @Override
+    public Map act(Redirector redirector, SourceResolver resolver, Map objectModel,
+                    String source, Parameters parameters) throws Exception
+    {
+        String journalName = parameters.getParameter(PARAM_JOURNAL_NAME);
+        if (journalName == null || journalName.length() == 0) return null;
+
+        // verify we have an accurate journal
+        Context context = ContextUtil.obtainContext(objectModel);
+        Concept journalConcept = JournalUtils.getJournalConceptByName(context,journalName);
```
Comment (ryscher) [inline dspace/modules/journal-landing/journal-landing-webapp/src/main/java/org/dspace/app/xmlui/aspect/journal/landing/ValidateRequest.java:45]: name lookup is here

**Pull request merged**